### ISSUE TITLE
Error handling for monthly mean History collection when run duration <1 month

### ISSUE THREAD

**Issue by lizziel** [GEOS-ESM/MAPL#1606]
I am encountering the following ESMF log error after modifying my HISTORY.rc file in GCHP 14.0 (uses MAPL 2.18.3) and reducing run duration to 1hr.

```
20220719 134413.379 ERROR            PET00 ESMCI_TimeInterval.C:1340 ESMCI::TimeInterval::operator/(t Invalid argument  - can't divide without startTime or endtime
20220719 134413.380 ERROR            PET00 ESMCI_TimeInterval.C:1340 ESMCI::TimeInterval::operator/(t Invalid argument  - can't divide without startTime or endtime
20220719 134414.927 INFO             PET00 ESMCI_Clock.C:2036 ESMCI::Clock::addAlarm() For clock ApplClock, alarm list is full (200 alarms), re-allocating to hold 400 alarms.
20220719 134415.252 INFO             PET00 ESMCI_Clock.C:2036 ESMCI::Clock::addAlarm() For clock ApplClock, alarm list is full (400 alarms), re-allocating to hold 600 alarms.
20220719 134415.260 INFO             PET00 ESMCI_Clock.C:2036 ESMCI::Clock::addAlarm() For clock ApplClock, alarm list is full (600 alarms), re-allocating to hold 800 alarms.
20220719 134415.267 INFO             PET00 ESMCI_Clock.C:2036 ESMCI::Clock::addAlarm() For clock ApplClock, alarm list is full (800 alarms), re-allocating to hold 1000 alarms.
20220719 134415.406 INFO             PET00 ESMCI_Clock.C:2036 ESMCI::Clock::addAlarm() For clock ApplClock, alarm list is full (1000 alarms), re-allocating to hold 1200 alarms.
20220719 134415.414 INFO             PET00 ESMCI_Clock.C:2036 ESMCI::Clock::addAlarm() For clock ApplClock, alarm list is full (1200 alarms), re-allocating to hold 1400 alarms.
20220719 134415.691 INFO             PET00 ESMCI_Clock.C:2036 ESMCI::Clock::addAlarm() For clock ApplClock, alarm list is full (1400 alarms), re-allocating to hold 1600 alarms.
20220719 134415.700 INFO             PET00 ESMCI_Clock.C:2036 ESMCI::Clock::addAlarm() For clock ApplClock, alarm list is full (1600 alarms), re-allocating to hold 1800 alarms.
20220719 134415.709 INFO             PET00 ESMCI_Clock.C:2036 ESMCI::Clock::addAlarm() For clock ApplClock, alarm list is full (1800 alarms), re-allocating to hold 2000 alarms.
20220719 134415.761 INFO             PET00 ESMCI_Clock.C:2036 ESMCI::Clock::addAlarm() For clock ApplClock, alarm list is full (2000 alarms), re-allocating to hold 2200 alarms.
```

I traced the issue to a new time-averaged collection that had monthly mean enabled. All other collections had monthly mean disabled. 

```
  StratBM.template:       '%y4%m2%d2_%h2%n2z.nc4',
  StratBM.format:         'CFIO',
  StratBM.timestampStart: .true.
  StratBM.monthly:        1
  StratBM.frequency:      010000
  StratBM.duration:       010000
  StratBM.mode:           'time-averaged'
  StratBM.fields:         'SpeciesConc_NO2 ', 'GCHPchem',
```

When I disable monthly mean in the new collection the issue goes away (no ESMF error log files are generated).

```
  StratBM.template:       '%y4%m2%d2_%h2%n2z.nc4',
  StratBM.format:         'CFIO',
  StratBM.timestampStart: .true.
  StratBM.monthly:        0
  StratBM.frequency:      010000
  StratBM.duration:       010000
  StratBM.mode:           'time-averaged'
  StratBM.fields:         'SpeciesConc_NO2 ', 'GCHPchem',
```

This error does not crash the run, but I wonder if there should be error handling in MAPL for the case of monthly mean turned on when duration is less than a month? If you would rather have it ignored, could you update it so that the ESMF error does not get triggered?

**Comment by mathomp4** [comment]
@lizziel I believe this is a bug (#1495) that we fixed with #1496 which is in MAPL 2.21 (put in 2.21.0 specifically).

I suppose the question for you is: can you update to MAPL 2.21? Or do you need to stay with the 2.18 releases? If so, we'd need to backport that patch for you into a 2.18.4, I guess.

(And I guess I'd want to consult with @atrayano if there are any other monthly collection fixes we've fixed since 2.18.3.)

**Comment by lizziel** [comment]
Ah, good to know! Clearly I have not been reading all the release notes. This isn't a big problem for us so I'll wait for the next MAPL upgrade which I'll probably do in the fall.
